### PR TITLE
Fix namespace handling check for Function/Procedure Overloading

### PR DESCRIPTION
**Pull request by roshan0708**
### Description
This commit fixes an issue where function and procedure overloading checks improperly included sys and pg_catalog schemas, affecting function and procedure creation when schemas was not explicitly specified. This change ensures that function and procedure overloading is handled correctly within the same schema and allows overloading across different schemas.

### Issues Resolved

Task: BABEL-5075

### Test Scenarios Covered ###
* **Use case based -** yes

* **Boundary conditions -** yes

* **Arbitrary inputs -**

* **Negative test cases -** yes

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/pltsql_utils.c [hunk 1]
```diff
@@ -208,15 +208,35 @@ pltsql_createFunction(ParseState *pstate, PlannedStmt *pstmt, const char *queryS
...
 
-			/* function, procedure */
+			/* If schema name is not specified, use the current default schema */
+			if (schemaname == NULL || !strlen(schemaname))
+			{
+				path_oids = fetch_search_path(false);
+				if (path_oids != NIL)
+				{
+					cur_schema_name = get_namespace_name(linitial_oid(path_oids));
+					if (cur_schema_name == NULL)
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/src/pltsql_utils.c]: Should we check if current schema is also valid?

File: contrib/babelfishpg_tsql/src/pltsql_utils.c [hunk 2]
```diff
@@ -208,15 +208,35 @@ pltsql_createFunction(ParseState *pstate, PlannedStmt *pstmt, const char *queryS
...
+					{
+						ereport(ERROR, (errcode(ERRCODE_UNDEFINED_SCHEMA),
+										errmsg("Current schema name could not be determined")));
+					}
```
Comment (thephantomthief) [inline contrib/babelfishpg_tsql/src/pltsql_utils.c:230]: Please add tests where we throw this error

File: contrib/babelfishpg_tsql/src/pltsql_utils.c [hunk 3]
```diff
@@ -208,15 +208,35 @@ pltsql_createFunction(ParseState *pstate, PlannedStmt *pstmt, const char *queryS
...
 
 			/* Get the schema name and function name. */
 			DeconstructQualifiedName(stmt->funcname, &schemaname, &funcname);
 
-			/* function, procedure */
+			/* If schema name is not specified, use the current default schema */
+			if (schemaname == NULL || !strlen(schemaname))
+			{
+				path_oids = fetch_search_path(false);
+				if (path_oids != NIL)
```
Comment (ahmed-shameem) [inline contrib/babelfishpg_tsql/src/pltsql_utils.c:223]: What if len(path_oids) > 1 ? In that case how do we decide? Is there any such case?

File: contrib/babelfishpg_tsql/src/pltsql_utils.c [hunk 4]
```diff
@@ -208,15 +208,35 @@ pltsql_createFunction(ParseState *pstate, PlannedStmt *pstmt, const char *queryS
...
+				{
+					cur_schema_name = get_namespace_name(linitial_oid(path_oids));
```
Comment (ahmed-shameem) [inline contrib/babelfishpg_tsql/src/pltsql_utils.c:225]: get_namespace_name returns p'alloced string, so let's free it after use.

File: test/JDBC/input/restricting_function_overloading-vu-verify.mix [hunk 5]
```diff
@@ -9,9 +9,6 @@ GO
 CREATE PROCEDURE OverloadedFunction AS BEGIN SELECT 1 END
```
Comment (ahmed-shameem) [inline test/JDBC/input/restricting_function_overloading-vu-verify.mix:9]: Let's add the test in latest/upgrade schedule if not already present.

**Review by ahmed-shameem — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/pltsql_utils.c (hunk 3 above)
Comment (roshan0708) [reply contrib/babelfishpg_tsql/src/pltsql_utils.c:223]: basically `path_oids` returns the list of oids of the all schemas present in the active search path and we only want the current schema oid for our case.

File: test/JDBC/input/restricting_function_overloading-vu-verify.mix (hunk 5 above)
Comment (roshan0708) [reply test/JDBC/input/restricting_function_overloading-vu-verify.mix:9]: This test file was already added by Shalini before, not sure about adding this in latest upgrade schedule file, if it was intentionally done or missed this.

**Review by thephantomthief — APPROVED** [review]

**Review by ahmed-shameem — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**